modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/resolver/ResolverSetUtils.java

```diff
@@ -188,82 +200,114 @@ private static ValueResolver getParameterValueResolver(String parameterName, Met
+              }
             }
+            resolverReference
+                .set(getValueResolverFor(parameterName, objectType, resolveAndinjectIfStatic(resolver, muleContext),
+                                         getDefaultValue(type),
+                                         getExpressionSupport(objectType), false, modelProperties, actualAcceptReferences,
+                                         valueResolverFactory));
+          } catch (MuleException e) {
+            throw new MuleRuntimeException(e);
           }
-          resolverReference
-              .set(getValueResolverFor(parameterName, objectType, resolveAndinjectIfStatic(resolver, muleContext),
-                                       getDefaultValue(type),
-                                       getExpressionSupport(objectType), false, modelProperties, actualAcceptReferences));
-        } catch (MuleException e) {
-          throw new MuleRuntimeException(e);
         }
-      }
 
-      @Override
-      protected void defaultVisit(MetadataType metadataType) {
-        resolverReference.set(getValueResolverFor(parameterName, metadataType, value, getDefaultValue(type),
-                                                  getExpressionSupport(metadataType), false, modelProperties, acceptsReferences));
-      }
+        @Override
+        protected void defaultVisit(MetadataType metadataType) {
+          resolverReference.set(getValueResolverFor(parameterName, metadataType, value, getDefaultValue(type),
+                                                    getExpressionSupport(metadataType), false, modelProperties,
+                                                    acceptsReferences, valueResolverFactory));
+        }
 
-      private ValueResolver getValueResolverFor(String parameterName, MetadataType metadataType, Object value,
-                                                Object defaultValue, ExpressionSupport expressionSupport, boolean required,
-                                                Set<ModelProperty> modelProperties, boolean acceptsReferences) {
-        return valueResolverFactory.of(parameterName, metadataType, value, defaultValue,
-                                       expressionSupport, required, modelProperties, acceptsReferences);
-      }
 
-    });
+
+      });
+    } else {
+      Object convertedValue = convertValueWithExpressionLanguage(value, type, muleContext.getExpressionManager());
+      resolverReference.set(getValueResolverFor(parameterName, type, convertedValue, null, NOT_SUPPORTED, false, modelProperties,
+                                                false, valueResolverFactory));
+    }
     initialiseIfNeeded(resolverReference.get(), muleContext);
     return resolverReference.get();
   }
 
+  private static Object convertValueWithExpressionLanguage(Object value, MetadataType type, ExpressionManager expressionManager) {
+    return expressionManager.evaluate("#[" + PAYLOAD + "]",
```

Comment:
either just use the literal `"#[payload]"` and let the compiler handle caches or use a constant with the entire expression, but this concatenation doesn't make sense